### DISCUSSION TITLE
pango.Layout.set_wrap support

### DISCUSSION THREAD

**Discussion by int-ua** [libvips/libvips#2073] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm creating 32x32px images with arbitrary text lengths, but the default pango layout is `WRAP_WORD` and it's not placing so much text in a significant number of cases where there is a short word at the beginning followed by a long word.

**Describe the solution you'd like**
The ability to change the wrap mode to `pango.WRAP_CHAR`:
https://developer.gnome.org/pygtk/stable/class-pangolayout.html#method-pangolayout--set-wrap

**Additional context**
https://github.com/libvips/pyvips/issues/204

**Comment by int-ua** [comment]
I was able to workaround it to a degree by using [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space):
```python
ZWSP = '\u200b'


def add_textbreaks(text, start=6):
    result = text[:start]
    for character in text[start:]:
        result += character + ZWSP
    return result
```
But (presumably) Pango adds hyphens and results are not satisfying:

Before | After
-- | --
![guidebook_dpi72](https://user-images.githubusercontent.com/1585431/91207091-e2df4f00-e710-11ea-8277-2ef75ed12071.png) | ![guidebook](https://user-images.githubusercontent.com/1585431/91207137-f8547900-e710-11ea-9b2c-3653c8682385.png)

If it's like this with `WRAP_CHAR` it will not help me and we can close this issue.

**Comment by jcupitt** [comment]
This is a good idea. Let's leave this open as an enhancement.

**Comment by jcupitt** [comment]
Updated Pango API link: https://docs.gtk.org/Pango/method.Layout.set_wrap.html

**Comment by jcupitt** [comment]
I made a PR, let's continue discussion there.

https://github.com/libvips/libvips/pull/3130
